packages/paragraph/src/withParagraph.ts

```diff
@@ -0,0 +1,23 @@
+import { PlateEditor, Value, isSelectionAtBlockEnd, isSelectionAtBlockStart } from '@udecode/plate-common';
+import { Editor, Path, move, removeNodes } from 'slate';
+
+
+
+
+export const withParagraph = <
+  V extends Value = Value,
+  E extends PlateEditor<V> = PlateEditor<V>
+>(
+  editor: E
+) => {
+  const { deleteForward, deleteBackward } = editor;
+  editor.deleteForward = (unit) => {
```

Comment:
This override will apply to all deletes forward, not just those inside paragraphs. If that's what we want (not necessarily a bad idea, but there may be edge cases we haven't considered, CC @zbeyens), then we should make this a separate plugin. You can find instructions for doing so in the CONTRIBUTING.md guide.

Otherwise, we can explicitly check that we're inside a paragraph.

Reply:
Once we figure out whether to keep this on paragraph or move it to its own plugin, can you add tests to verify the correct behaviour in all relevant cases?